### PR TITLE
User API endpoints

### LINKED ISSUE #2077: Support User API endpoints
A few user endpoints have been adding server-side over the past few months. Would be good to support them natively in `huggingface_hub`, especially for Spaces that rely on social feats.

Here is a list of them:
- [ ] `/overview` (e.g. https://huggingface.co/api/users/julien-c/overview). Returns details about a user on the Hub: user id, name, profile, avatar url, is_pro, nb_models, nb_datasets, ..., nb_likes, etc.
- [ ] `/likes` (e.g. https://huggingface.co/api/users/julien-c/likes). Returns all repos liked by a user.
- [ ] `/followers` (e.g. https://huggingface.co/api/users/julien-c/followers). Returns all followers of a user.
- [ ] `/following` (e.g. https://huggingface.co/api/users/julien-c/following). Returns all users followed by a user.
- [ ] `/api/organizations/:name/members` (e.g. https://huggingface.co/api/organizations/huggingface/members). List of member of an org.
  
Each endpoint can be added as a separate PR. Return type should always be a dataclass. Might also be good to take the opportunity to document those endpoints in [api.md](https://github.com/huggingface/hub-docs/blob/main/docs/hub/api.md) (separate repo).

More priority on the `/overview` and `/members` endpoints. Lower prio for the others.

### PR DESCRIPTION
**Pull request by bilgehanertan**
User API endpoints are implemented. 

`/likes` endpoint was already implemented on `hf_api.py:list_liked_repos:2078`. 

Resolves #2077

### CODE REVIEW

**Comment by bilgehanertan** [comment]
> Hey @bilgehanertan thanks a lot for your first contribution! This is really nicely implemented and documented 🔥
> 
> I just reviewed your PR. There are quite some comments but most of them are really minimal (just a matter of how to parse the data). Once those changes are done, I think we should be good to merge :)

Thank you! I will be changing them today.

**Comment by bilgehanertan** [comment]
> Hey @bilgehanertan thanks a lot for your first contribution! This is really nicely implemented and documented 🔥
> 
> I just reviewed your PR. There are quite some comments but most of them are really minimal (just a matter of how to parse the data). Once those changes are done, I think we should be good to merge :)

I fixed them all 👍

File: src/huggingface_hub/hf_api.py [hunk 1]
```diff
@@ -1197,12 +1197,58 @@ class User:
...
+            Whether the authenticated user is following this user.
+        details (`str`, *optional*):
+            User's details.
     """
 
     # Metadata
     avatar_url: str
     username: str
     fullname: str
+    num_models: Optional[int] = None
```
Comment (Wauplin) [inline src/huggingface_hub/hf_api.py:1224]: I'm sorry I made a mistake in https://github.com/huggingface/huggingface_hub/pull/2147#discussion_r1541223989. Would it be possible to add back `is_pro: Optional[bool] = None` to the list. I only wanted to select `user_type` as the field to remove. Thanks in advance!

(also update `__init__.py` and docstring accordingly)

**Review by Wauplin — COMMENTED** [review]
Nice! :fire: Really great job, thanks @bilgehanertan! I just realized I made a mistake in a comment yesterday. Would it be possible to add back `is_pro` as attribute? Apart from that, it's ready to be merged :) Thanks in advance!

File: src/huggingface_hub/hf_api.py (hunk 1 above)
Comment (bilgehanertan) [reply src/huggingface_hub/hf_api.py:1227]: @Wauplin Fixed it!

**Review by Wauplin — APPROVED** [review]
Perfect! Thanks again @bilgehanertan! I'll merge as soon as the CI is green :)

**Comment by Wauplin** [comment]
Merged it as failing test was unrelated.

**Pull request merged**